### PR TITLE
Update the test suite to use Pytest

### LINKED ISSUE #28: Update tests
On the one hand, the tests are no longer up to date with the new features (join types). On the other hand, the code has to be tested and adapted with the HAPI API.

### PR DESCRIPTION
**Pull request by SaulLu**
fix #28 and #29

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -2,56 +2,56 @@
...
-    os.chdir(
-        os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-    )
-    test_1()
-    test_2()
-    test_3()
-    test_4()
-    test_5()
+    os.chdir(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    test_resources_in_dataframe("tests/1", "http://hapi.fhir.org/baseR4/")
```
Comment (LaRiffle) [inline tests/test_examples.py:57]: With pytest, i don't think you need to explicitely call `test_resources_in_dataframe` it searchs automatically for all functions `def test_*`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -34,7 +37,7 @@ jobs:
     if: github.ref == 'refs/heads/master'
     strategy:
       matrix:
-        python-version: ["3.7", "3.8"]
+        python-version: ["3.8"]
```
Comment (LaRiffle) [inline .github/workflows/main.yml:40]: Are we sure we want to remove 3.7 tests ?

File: tests/test_examples.py (hunk 1 above)
Comment (SaulLu) [reply tests/test_examples.py:57]: Indeed, this call in the __name__ == "__main__" is not necessary for `pytest` but I had put it if ever we wanted to launch the function as a classic script. Do you want me to delete it?

File: .github/workflows/main.yml (hunk 2 above)
Comment (SaulLu) [reply .github/workflows/main.yml:40]: The python version of line 40 was only used to publish on PyPI the new version of the module. So, if the module is published twice (i.e. with both python versions), there is, unfortunately, an error, that's why I removed it. Did I correctly understand how the workflows worked?

File: tests/test_examples.py (hunk 1 above)
Comment (LaRiffle) [reply tests/test_examples.py:57]: I'd say the tests should from on now always be launched with pytest

File: .github/workflows/main.yml (hunk 2 above)
Comment (LaRiffle) [reply .github/workflows/main.yml:40]: Ah I see! In that case let's not use a matrix to choose python versions but smthg like this: https://github.com/OpenMined/PySyft/blob/master/.github/workflows/publish.yml#L12
(just for the python version selection)

File: tests/test_examples.py (hunk 1 above)
Comment (SaulLu) [reply tests/test_examples.py]: got it! i deleted it in a new commit.

File: .github/workflows/main.yml (hunk 2 above)
Comment (SaulLu) [reply .github/workflows/main.yml]: I changed it. Are the changes right for you?

**Pull request merged**